content/knowledge-codemagic/build-inputs.md

```diff
@@ -0,0 +1,200 @@
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  greetings:
+    inputs:
+      name:
+        description: Who is greeted?
+        default: Codemagic
+    scripts:
+      - echo "Hello, ${{ inputs.name }}"
+{{< /highlight >}}
+
+## Starting builds with Inputs
+
+All inputs must be specified to successfully start a build, either by providing a `default` value in the YAML configuration, or giving a one-off value when starting the build. 
+
+{{<notebox>}}
+**Note**: Builds will fail if invalid values are provided (strings for number inputs, undefined choice options, etc.) or values are missing.
+{{</notebox>}}
+
+### Starting builds manually via Codemagic UI
+
+When starting a build via the Codemagic UI, you will automatically be prompted to enter the inputs. Inputs that have predefined default values will be prefilled with those values from configuration file. All other inputs must be manually entered before the build can be started.
+
+{{<notebox>}}
+**Note on inputs without default values:**
+
+Not entering anything for a string input will result in an empty string, i.e. `""`. Boolean inputs default to `false` in the UI. For other input types, an actual value that matches the requested type must be entered.
+{{</notebox>}}
+
+### Starting builds using webhook events
+
+Only workflows that do not require user input for values can be started with webhook events. If you want to use Git events or scheduled builds to automatically trigger builds for workflows with inputs, ensure that all inputs in those workflows have default values. Otherwise, the build will fail due to undefined inputs.
+
+## YAML schema for inputs
+
+Build inputs are defined in `codemagic.yaml` as a mapping `workflow.<workflow_id>.inputs` where keys are input IDs and values are inputs that have the following fields.
+
+### `description`
+
+**Required**. A **string** description for this build input. Description is displayed in Codemagic when manually starting a build for this workflow and user is prompted to provide values for the inputs.
+
+### `type`
+
+**This must be one of: `boolean`, `choice`, `number` or `string`. By default, `string` is assumed.**
+
+Defines the data type of the input parameter. Input values for the `choice` type are resolved to strings and must be defined in the `options` field. Values for inputs with types `boolean` and `number` are persisted as booleans and integers or floating point numbers respectively, instead of converting them to strings as long as they are not directly used in string interpolations.
+
+### `options`
+
+Provide a list of values as options for the `choice` input. Values are all implicitly cast to strings. **Required if type is `choice` and prohibited otherwise**.
+
+### `default`
+
+Provide a default value for the input parameter. If `type` is choice, then it must be one of the defined `options`, otherwise value type must match with the `type` definition.
```

Comment:
```suggestion
Provide a default value for the input parameter. If type is `choice`, then it must be one of the defined `options`, otherwise value type must match with the `type` definition.
```